bluebutton-data-model-codegen/src/main/java/gov/hhs/cms/bluebutton/data/model/codegen/RifLayoutsProcessor.java

```diff
@@ -488,6 +505,30 @@ private TypeSpec generateHeaderEntity(MappingSpec mappingSpec) throws IOExceptio
 			headerEntityClass.addMethod(headerFieldSetter.build());
 		}
 
+		/*
+		 * Create an Entity field for additional database fields that we need to store
+		 * data for whereas there isn't a corresponding RIF input field.
+		 */
+		for (RifField addlDatabaseField : mappingSpec.getHeaderEntityAdditionalDatabaseFields()) {
+			FieldSpec headerField = FieldSpec
+					.builder(selectJavaFieldType(addlDatabaseField), addlDatabaseField.getJavaFieldName(),
+							Modifier.PRIVATE)
+					.addAnnotations(createAnnotations(mappingSpec, addlDatabaseField)).build();
+			headerEntityClass.addField(headerField);
+			  
+			MethodSpec.Builder headerFieldGetter = MethodSpec.methodBuilder(calculateGetterName(headerField))
+					.addModifiers(Modifier.PUBLIC).returns(selectJavaPropertyType(addlDatabaseField));
+			addGetterStatement(addlDatabaseField, headerField, headerFieldGetter);
+			headerEntityClass.addMethod(headerFieldGetter.build());
+
+			MethodSpec.Builder headerFieldSetter = MethodSpec.methodBuilder(calculateSetterName(headerField))
+					.addModifiers(Modifier.PUBLIC).returns(void.class)
+					.addParameter(selectJavaPropertyType(addlDatabaseField), headerField.name);
+			addSetterStatement(addlDatabaseField, headerField, headerFieldSetter);
+			headerEntityClass.addMethod(headerFieldSetter.build());
+
+		}
```

Comment:
I'm stoked that you were able to jump into this code and hack on it -- I was worried that it was too much of a mess. :guitar: